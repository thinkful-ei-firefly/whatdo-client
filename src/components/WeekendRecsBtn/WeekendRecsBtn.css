.weekend__button__component {
  margin: 0;
  padding: 0;
  width: 100%;
  height: 10vh;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  /* background: #000; */
}

.weekend__button {
  width: 220px;
  height: 50px;
  border: none;
  outline: none;
  color: #fff;
  background: rgb(0, 64, 143);
  cursor: pointer;
  position: relative;
  z-index: 0;
  font-size: large;
  border-radius: 10px;
}

.weekend__button:before {
  content: "";
  background: linear-gradient(
    45deg,
    #ff0000,
    #ff7300,
    #fffb00,
    #48ff00,
    #00ffd5,
    #002bff,
    #7a00ff,
    #ff00c8,
    #ff0000
  );
  position: absolute;
  top: -2px;
  left: -2px;
  background-size: 400%;
  z-index: -1;
  filter: blur(5px);
  width: calc(100% + 4px);
  height: calc(100% + 4px);
  animation: glowing 20s linear infinite;
  opacity: 0;
  transition: opacity 0.3s ease-in-out;
  border-radius: 10px;
}

.weekend__button:hover {
  color: #000;
}

.weekend__button:hover:after {
  background: transparent;
}

.weekend__button:before {
  opacity: 1;
}

.weekend__button:after {
  z-index: -1;
  content: "";
  position: absolute;
  width: 100%;
  height: 100%;
  background: rgba(65, 0, 97, 0.906);
  left: 0;
  top: 0;
  border-radius: 10px;
}

@keyframes glowing {
  0% {
    background-position: 0 0;
  }
  50% {
    background-position: 400% 0;
  }
  100% {
    background-position: 0 0;
  }
}
