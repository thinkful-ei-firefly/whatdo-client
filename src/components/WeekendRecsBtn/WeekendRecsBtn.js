import React from 'react'
import SearchContext from '../../contexts/SearchContext'
import GeocodingApiService from '../../services/geocoding-api-service'
import './WeekendRecsBtn.css'

export default class WeekendRecsBtn extends React.Component {

  static contextType = SearchContext

  state = { zip: '' }

  getLocation = async () => {
    this.setState({loading: true})

    const options = {
      timeout: 5000,
      enableHighAccuracy: true,
      maximumAge: 0
    }

    const success = async (pos) => {
      const {latitude, longitude} = pos.coords
      try{
        const res = await GeocodingApiService.getZip(latitude, longitude)
        this.setState({zip: res.results[0].address_components[0].long_name})
      } catch(err){
        console.log(err.message)
      }
      this.setState({loading: false})
    }

    const error = (err) => {
      console.log(err)
      this.setState({loading: false})
    }

    // eslint-disable-next-line no-unused-vars
    navigator.geolocation.getCurrentPosition((pos) => success(pos), error, options)
  }

  handleClick = async () => {
    //starting with today, step through dates until we hit the next Saturday or Sunday
    let date = new Date(Date.now())
    while (date.getDay() > 0 && date.getDay() < 6) {
      date.setTime(date.getTime()+86400000)
    }
    date = date.toISOString().slice(0,10)

    //get the favorite categories of the user
    const userFavorites = this.context.categories || 'music,sports,outdoors_recreation,attractions,performing_arts,comedy,food,singles_social,festivals_parades,holiday'
    
    //get the current/registered zip code of the user
    const zipCode = this.state.zip || this.context.zipCode
    
    //pass everything to the search
    this.context.apiSearch(zipCode, 15, date, userFavorites)

    //redirect to the results page
    this.props.redirectToEvents()

  }

  componentDidMount() {
    this.getLocation()
  }

  render () {
    return (
      <div>
        <button className="weekend_button" onClick={this.handleClick}>What Do this Weekend?</button>
      </div>
    );
  }
}