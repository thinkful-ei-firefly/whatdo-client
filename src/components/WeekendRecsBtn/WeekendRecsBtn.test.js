import React from 'react'
import WeekendRecsBtn from './WeekendRecsBtn'
import renderer from 'react-test-renderer'

describe('<WeekendRecsBtn />', () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(
        <WeekendRecsBtn />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})