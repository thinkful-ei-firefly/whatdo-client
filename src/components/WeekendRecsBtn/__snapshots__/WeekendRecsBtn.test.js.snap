// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeekendRecsBtn /> renders as expected 1`] = `
<div
  className="weekend__button__component"
>
  <button
    className="weekend__button"
    onClick={[Function]}
  >
    What Do this Weekend?
  </button>
</div>
`;
